src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java

```diff
@@ -764,6 +755,53 @@ public RegisteredUser resetPassword(final String token, final String newPassword
         authenticator.setOrChangeUsersPassword(user, newPassword);
         return user;
     }
+
+    /**
+     *  Create and save an account deletion token for the specified user.
+     *
+     * @param user - the user requesting deletion.
+     * @return - the deletion token to send to the user.
+     * @throws SegueDatabaseException - on database error.
+     */
+    public AccountDeletionToken createAccountDeletionTokenForUser(final RegisteredUserDTO user) throws SegueDatabaseException {
+
+        String token;
+        try {
+            IPasswordAuthenticator authenticator = (IPasswordAuthenticator) this.registeredAuthProviders
+                    .get(AuthenticationProvider.SEGUE);
+
+            token = authenticator.createAccountDeletionTokenSecret();
+        } catch (NoSuchAlgorithmException e) {
+            throw new SegueDatabaseException("Failed to create account deletion token", e);
+        }
+
+        // Valid for 7 days:
```

Comment:
I don't like the proliferation of config properties for things we never change. Most places that use config properties also load them once on class instantiation and then the only way to change them is to restart the API, which is equivalent to doing a release to change the value anyway.
I am happy to review the value though!

Reply:
> Most places that use config properties also load them once on class instantiation and then the only way to change them is to restart the API

I thought we had an admin endpoint to reload the config at runtime, although admittedly I haven't checked where that will and won't work. 

Being able to inject a config with a short expiry time could be useful for tests, but you could probably achieve the same thing by updating the test data to include a pre-expired token, so I don't mind.

Reply:
We do, but it can't fix places that use `@Named` or get the property and store it as an attribute; it will only work if they do `properties.getProperty()` every time they use it. It's about a 50-50 mix of approaches, I think. Maybe that's a refactor for the future!

I agree that for testing the won't affect anything because we can just make a token older than the limit, and for local dev the overhead in changing the code versus the config file is minimal.